### ISSUE TITLE
definition of "TOKEN #foo:" results in NPE

### ISSUE THREAD

**Issue by manticore-projects** [javacc21/javacc21#48]
Shortened Example:
```
TOKEN #KeyWordToken: /* SQL Keywords. prefixed with K_ to avoid name clashes */
{
    <K_ACTION: "ACTION">
|   <K_ADD:"ADD">
...
|   <K_YAML:"YAML">
|   <K_ZONE:"ZONE">
}
```

Exception in thread "main" java.lang.NullPointerException
	at com.javacc.parsegen.ParserData.semanticize(ParserData.java:244)
	at com.javacc.Grammar.semanticize(Grammar.java:312)
	at com.javacc.Main.mainProgram(Main.java:286)
	at com.javacc.Main.main(Main.java:267)

I think, that:
1) the NPE should be caught properly and
2) I should see a message, what is wrong with my Grammar file and where

**Comment by revusky** [comment]
Well, this is surely a bug. It's strange because, as I pointed out to you, I'm using this in internal development. But there is obviously a problem here. 

Could you send me, maybe by email the current grammar where you're getting this bug.

**Comment by manticore-projects** [comment]
Thank you for prompt response.
Please see the Grammar File attached.
[JSqlParserCC.zip](https://github.com/javacc21/javacc21/files/6493658/JSqlParserCC.zip)

**Comment by revusky** [comment]
Okay, I think this is fixed now.

**Comment by manticore-projects** [comment]
Thank you for the prompt solution, I have tested it successfully.
Fixed by commit 51d96fb25cca9361fa99d9141b0a6d8c1d372a67
